test/test.cpp

```diff
@@ -36,7 +36,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 unit_test_t _g_unit_tests[1024];
 int _g_num_unit_tests = 0;
-int _g_test_failures = 0;
+int _g_test_failures = 0; // flushed at start of every unit
+int _g_test_total_failures = 0;
```

Comment:
did this used to return incorrect return code from the test process? if the last test passed, would it shadow previous failures?
Is there a good reason to clear ``_g_test_failures`` each turn through the loop? I suppose there needs to be some way to tell whether a test had failures or not. an alternative way would be to copy this counter at the beginning of the test and then compare at the end. I think that would be a bit nicer, as it would require one less global variable.